src/types.jl

```diff
@@ -235,317 +235,281 @@ mutable struct ColorTable
-    @convert(
-        GDALDataType::ImageCore.Normed,
-        GDT_Byte::ImageCore.N0f8,
-        GDT_UInt16::ImageCore.N0f16,
-        GDT_UInt32::ImageCore.N0f32,
-    )
+@convert(
+    GDALDataType::ImageCore.Normed,
+    GDT_Byte::ImageCore.N0f8,
+    GDT_UInt16::ImageCore.N0f16,
+    GDT_UInt32::ImageCore.N0f32,
 )
 
-eval(
-    @convert(
-        GDALDataType::DataType,
-        GDT_Unknown::Any,
-        GDT_Byte::UInt8,
-        GDT_UInt16::UInt16,
-        GDT_Int16::Int16,
-        GDT_UInt32::UInt32,
-        GDT_Int32::Int32,
-        GDT_Float32::Float32,
-        GDT_Float64::Float64,
-        GDT_CFloat32::ComplexF32,
-        GDT_CFloat64::ComplexF64
-    )
+@convert(
+    GDALDataType::DataType,
+    GDT_Unknown::Any,
+    GDT_Byte::UInt8,
+    GDT_UInt16::UInt16,
+    GDT_Int16::Int16,
+    GDT_UInt32::UInt32,
+    GDT_Int32::Int32,
+    GDT_Float32::Float32,
+    GDT_Float64::Float64,
+    GDT_CFloat32::ComplexF32,
+    GDT_CFloat64::ComplexF64
 )
 
-eval(
-    @convert(
-        OGRFieldType::GDAL.OGRFieldType,
-        OFTInteger::GDAL.OFTInteger,
-        OFTIntegerList::GDAL.OFTIntegerList,
-        OFTReal::GDAL.OFTReal,
-        OFTRealList::GDAL.OFTRealList,
-        OFTString::GDAL.OFTString,
-        OFTStringList::GDAL.OFTStringList,
-        OFTWideString::GDAL.OFTWideString,
-        OFTWideStringList::GDAL.OFTWideStringList,
-        OFTBinary::GDAL.OFTBinary,
-        OFTDate::GDAL.OFTDate,
-        OFTTime::GDAL.OFTTime,
-        OFTDateTime::GDAL.OFTDateTime,
-        OFTInteger64::GDAL.OFTInteger64,
-        OFTInteger64List::GDAL.OFTInteger64List,
-    )
+@convert(
+    OGRFieldType::GDAL.OGRFieldType,
+    OFTInteger::GDAL.OFTInteger,
+    OFTIntegerList::GDAL.OFTIntegerList,
+    OFTReal::GDAL.OFTReal,
+    OFTRealList::GDAL.OFTRealList,
+    OFTString::GDAL.OFTString,
+    OFTStringList::GDAL.OFTStringList,
+    OFTWideString::GDAL.OFTWideString,
+    OFTWideStringList::GDAL.OFTWideStringList,
+    OFTBinary::GDAL.OFTBinary,
+    OFTDate::GDAL.OFTDate,
+    OFTTime::GDAL.OFTTime,
+    OFTDateTime::GDAL.OFTDateTime,
+    OFTInteger64::GDAL.OFTInteger64,
+    OFTInteger64List::GDAL.OFTInteger64List,
 )
 
-eval(
-    @convert(
-        OGRFieldType::DataType,
-        OFTInteger::Bool,
-        OFTInteger::Int16,
-        OFTInteger::Int32,  # default type comes last
-        OFTIntegerList::Vector{Int32},
-        OFTReal::Float32,
-        OFTReal::Float64,  # default type comes last
-        OFTRealList::Vector{Float64},
-        OFTString::String,
-        OFTStringList::Vector{String},
-        OFTWideString::Nothing,
-        OFTWideStringList::Nothing,
-        OFTBinary::Vector{UInt8},
-        OFTDate::Dates.Date,
-        OFTTime::Dates.Time,
-        OFTDateTime::Dates.DateTime,
-        OFTInteger64::Int64,
-        OFTInteger64List::Vector{Int64},
-    )
+@convert(
+    OGRFieldType::DataType,
+    OFTInteger::Bool,
+    OFTInteger::Int16,
+    OFTInteger::Int32,  # default type comes last
```

Comment:
(optional) Given that we're using dictionaries for representing these mappings, it has always made me a little uncomfortable that there's no good way of asserting "default type comes last" for mappings that are not bijective.
* Should it be included in the docstring?
* Should we have unit tests for them?
* Should we organize the implementation such that it becomes clear how we're enforcing this by construction?

Reply:
After thinking more about it, I realized these are requests (outside of the scope of this PR) rather than questions, so I've opened an issue for it in #232 instead.